prog/vm/nexus.rb

```diff
@@ -293,8 +293,26 @@ def before_run
   label def run
     vm.nics.each { _1.incr_setup_nic }
     host.sshable.cmd("sudo systemctl start #{q_vm}")
+    hop_wait_sshable
+  end
+
+  label def wait_sshable
+    addr = vm.ephemeral_net4 || vm.ephemeral_net6.nth(2)
+    out = `ssh -o BatchMode=yes -o ConnectTimeout=1 -o PreferredAuthentications=none user@#{addr} 2>&1`
+    if out.include? "Host key verification failed."
+      vm.update(display_state: "running")
+      Clog.emit("vm provisioned") { {vm: vm.values, provision: {vm_ubid: vm.ubid, vm_host_ubid: host.ubid, duration: Time.now - vm.created_at}} }
+      hop_create_billing_record
+    end
+    nap 1
+  end
+
+  label def create_billing_record
+    project = vm.projects.first
+    hop_wait unless project.billable
```

Comment:
I think we can move this check to wait_sshable (i.e. we would directly hop to wait skipping create_billing_record)

Reply:
I believe that checking it as the first step of the skipped label enhances clarity and maintainability.

https://github.com/ubicloud/ubicloud/pull/96#issuecomment-1524732612

Reply:
fair enough. I'm not sure which way is better, but I don't have strong preference either.